File: app/src/main/java/jp/ac/dendai/c/jtp/adventuresample/scene/GameState.java
package jp.ac.dendai.c.jtp.adventuresample.scene;
public enum GameState {
    first(new First()),second(new Second()),
    ending(new Ending()), badend(new BadEnd())
    ;
    private Scene scene;
    public Scene getScene(){
        return scene;
    }
    private GameState(Scene s){
        scene = s;
    }
    public static Scene getInitialScene(){
        return first.getScene();
    }
}

File: app/src/main/java/jp/ac/dendai/c/jtp/adventuresample/scene/GameState.kt
package jp.ac.dendai.c.jtp.adventuresample.scene

enum class GameState private constructor(val scene: Scene) {
    first(First()), second(Second()),
    ending(Ending()), badend(BadEnd()), trueend(TrueEnd());


    companion object {
        val initialScene: Scene
            get() = first.scene
    }
}